src/logger/src/logger.rs

```diff
@@ -1,776 +1,343 @@
-        let mut guard = extract_guard(self.instance_id.write());
-        *guard = instance_id;
-        self
+        Ok(())
     }
+}
 
-    /// Explicitly sets the max log level for the Logger.
-    /// The default level is WARN. So, ERROR and WARN statements will be shown (i.e. all that is
-    /// bigger than the level code).
-    ///
-    /// # Arguments
-    ///
-    /// * `level` - Set the highest log level.
-    /// # Example
-    ///
-    /// ```
-    /// use std::ops::Deref;
-    ///
-    /// use logger::{info, warn, LOGGER};
-    ///
-    /// let l = LOGGER.deref();
-    /// l.set_max_level(log::LevelFilter::Warn);
-    /// assert!(l.configure(Some("MY-INSTANCE".to_string())).is_ok());
-    /// info!("An informational log message");
-    /// warn!("A test warning message");
-    /// ```
-    /// The code above will more or less print:
-    /// ```bash
-    /// 2018-11-07T05:34:25.180751152 [MY-INSTANCE:INFO:logger/src/lib.rs:389] A test warning
-    /// message
-    /// ```
-    pub fn set_max_level(&self, level: LevelFilter) -> &Self {
-        set_max_level(level);
-        self
-    }
+#[derive(Debug)]
+pub struct LogFiler {
+    pub level: log::LevelFilter,
+}
+#[derive(Debug)]
+pub struct LogFormat {
+    pub show_level: bool,
+    pub show_log_origin: bool,
+}
+#[derive(Debug)]
+pub struct LoggerConfiguration {
+    pub target: Option<std::fs::File>,
+    pub filter: LogFiler,
+    pub format: LogFormat,
+}
+#[derive(Debug)]
+pub struct Logger(pub Mutex<LoggerConfiguration>);
 
-    /// Get the current thread's name.
-    fn get_thread_name(&self) -> String {
-        thread::current().name().unwrap_or("-").to_string()
+impl Log for Logger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        metadata.level() <= self.0.lock().unwrap().filter.level
     }
 
-    /// Creates the first portion (to the left of the separator)
-    /// of the log statement based on the logger settings.
-    fn create_prefix(&self, record: &Record) -> String {
-        let mut prefix: Vec<String> = vec![];
-
-        let instance_id = extract_guard(self.instance_id.read());
-        if !instance_id.is_empty() {
-            prefix.push(instance_id.to_string());
-        }
-
-        // Attach current thread name to prefix.
-        prefix.push(self.get_thread_name());
-
-        if self.show_level() {
-            prefix.push(record.level().to_string());
-        };
-
-        if self.show_file_path() {
-            prefix.push(record.file().unwrap_or("unknown").to_string());
-        };
+    fn log(&self, record: &Record) {
+        // Lock the logger.
+        let mut guard = self.0.lock().unwrap();
 
-        if self.show_line_numbers() {
-            if let Some(line) = record.line() {
-                prefix.push(line.to_string());
+        // Check if the log message is enabled
+        {
+            let enabled = record.level() <= guard.filter.level;
+            if !enabled {
+                return;
```

Comment:
This should already be checked by the `log!` macro, via a call to `enabled`, no?

Reply:
It is not via `enabled` but you are correct it is not needed, see https://github.com/firecracker-microvm/firecracker/pull/4047#discussion_r1304159168.